lib/SILGen/SILGenLValue.cpp

```diff
@@ -2788,6 +2810,22 @@ LValue SILGenLValue::visitRec(Expr *e, SGFAccessKind accessKind,
   if (e->getType()->is<LValueType>() || e->isSemanticallyInOutExpr()) {
     return visitRecInOut(*this, e, accessKind, options, orig);
   }
+  
+  // If the base is a load of a noncopyable type (or, eventually, when we have
+  // a `borrow x` operator, the operator is used on the base here), we want to
+  // apply the lvalue within a formal access to the original value instead of
+  // an actual loaded copy.
```

Comment:
So we do have the expression form `_borrow x` behind `-enable-experimental-move-only`, so it's worth ensuring this also works for that too. This:

```
struct X: ~Copyable {
  var something: Int { return 1 }
}

func f() -> Int {
  var x = X()
  return (_borrow x).something
}
```

gives an AST like this:

```
(member_ref_expr ...
  (paren_expr ...
    (load_expr implicit type='X' ...
      (borrow_expr type='@lvalue X' ...
        (declref_expr type='@lvalue X' ... )))))
```

It doesn't appear that SILGenLValue has a visitor for a `BorrowExpr` yet, so this might crash now (haven' tried your branch yet).